### DISCUSSION TITLE
Anhänge

### DISCUSSION THREAD

**Discussion by BlueStarHH** [LandrixSoftware/XRechnung-for-Delphi#25] [Q&A]
Ich erstelle eine ZUGFeRD-Rechnung im Profil XRechnung. Es sollten Dateien angehängt werden. Müssen diese direkt in die PDF-Datei oder in in den den XML-Teil eingebettet werden?

Wenn ich Anhänge in den XML-Teil einbetten möchte, werden diese Anhänge nicht geschrieben, da in der unit intf.XRechnung_3_0.pas in der Funktion TXRechnungInvoiceAdapter301.SaveDocumentUNCEFACT diese if-Abfrage vorhanden ist:

```
if _Invoice.ProjectReference <> '' then
  for i := 0 to _Invoice.Attachments.Count -1 do
```

Warum muss eine ProjectReference vorhanden sein, damit Anhänge gespeichert werden können?

**Comment by landrix** [comment]
erledigt
